assets/app/view/part/city.rb

```diff
@@ -145,13 +152,70 @@ def render_part
         children = []
         children << render_box(slots.size) if slots.size.between?(2, 6)
         children.concat(slots)
+
+        if (revenue = render_revenue)
+          children << revenue
+        end
+
         props = @city.solo? ? {} : { on: { click: -> { touch_node(@city) } } }
         h(:g, props, children)
       end
 
-      # TODOS:
-      # - do actual math and get points for the 3-slot hexagon, rather than
-      #   scaling the full-size hexagon
+      def render_revenue
+        revenues = @city.revenue.values.uniq
+        return if revenues.uniq.size > 1
+
+        revenue = revenues.first
+        return if revenue.zero?
+
+        # let View::Tile worry about rendering revenue if there are too many
+        # individual cities (eg, Chi in 1846)
+        return if @tile.cities.size > 2
```

Comment:
don't you have @cities loaded? maybe rename that to num_cities and use that in both places